compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5471,8 +5471,11 @@ static TR::Register *inlineArrayCmpP10(TR::Node *node, TR::CodeGenerator *cg, bo
    TR::LabelSymbol *resultLabel = generateLabelSymbol(cg);
 
    bool is64bit = cg->comp()->target().is64Bit();
+   bool isLengthNode64bit = lengthNode->getDataType().isInt64();
+   TR_ASSERT_FATAL_WITH_NODE(lengthNode, lengthNode->getDataType().isInt64() || lengthNode->getDataType().isInt32(),
+                             "lengthNode not int64 or int32. Datatype: %s", node->getDataType().toString());
 
-   if (isArrayCmpLen && !is64bit)
+   if (!is64bit && isLengthNode64bit)
```

Comment:
from Spencer comment, it seems impossible to have 64bit length node on 32bit platforms. we have no need to handle this combination.  plus, it is functionally useless to have that combination.  furthermore, it adds complexity to handle register-pair for long on 32bit platforms.

Reply:
It is possible to have a 64bit length node on 32bit. I am able to get trees like this on 32bit AIX:
```
n524n     iRegStore gr2                                                                       [0xef0972c0] bci=[-1,14,4] rc=0 vc=1494 vn=- li=4 udi=- nc=1
n521n       l2i                                                                               [0xef097200] bci=[-1,10,4] rc=3 vc=1494 vn=- li=4 udi=- nc=1
n520n         arraycmplen  <arraycmplen>[#244  helper Method] [flags 0x400 0x0 ] ()           [0xef0971c0] bci=[-1,10,4] rc=1 vc=1494 vn=- li=4 udi=- nc=3 flg=0x20
n509n           aiadd (X>=0 internalPtr )                                                     [0xef096f00] bci=[-1,10,4] rc=1 vc=1494 vn=- li=4 udi=- nc=2 flg=0x8100
n546n             ==>aRegLoad
n511n             iconst 8 (X!=0 X>=0 )                                                       [0xef096f80] bci=[-1,10,4] rc=2 vc=1494 vn=- li=4 udi=- nc=0 flg=0x104
n514n           aiadd (X>=0 internalPtr )                                                     [0xef097040] bci=[-1,13,4] rc=1 vc=1494 vn=- li=4 udi=- nc=2 flg=0x8100
n545n             ==>aRegLoad
n511n             ==>iconst 8
n519n           iu2l (highWordZero X>=0 )                                                     [0xef097180] bci=[-1,4,3] rc=1 vc=1494 vn=- li=4 udi=- nc=1 flg=0x4100
n13n              ==>iloadi
```
The length node is a 64bit value in this case.

Reply:
> but IdiomTransformations will insert an i2l on the length child on 64-bit platforms. (quoted from above) 

please confirm it with @Spencer-Comin.